src/transformers/models/maest/feature_extraction_maest.py

```diff
@@ -0,0 +1,148 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Feature extractor class for Music Audio Efficient Spectrogram Transformer.
+"""
+
+
+import numpy as np
+import torch
+from librosa.feature import melspectrogram
+
+from ...utils import logging
+from ..audio_spectrogram_transformer import ASTFeatureExtractor
+
+
+logger = logging.get_logger(__name__)
+
+
+class MAESTFeatureExtractor(ASTFeatureExtractor):
```

Comment:
Rather than subclassing `ASTFeatureExtractor`, it's probably cleaner if you just use `# Copied from` for each class that is a direct copy from AST, and override the classes that are new

Reply:
Addressed in 9ce296f